lib/ssl/src/tls_connection.erl

```diff
@@ -1164,6 +1168,8 @@ next_tls_record(Data, StateName,
             _ ->
                 State0#state.connection_env#connection_env.negotiated_version
         end,
+    #{current_write := #{max_fragment_length := MaxFragLen}} = State0#state.connection_states,
+    SslOpts = SslOpts0#{max_fragment_length => MaxFragLen},
```

Comment:
I wonder if we can avoid setting `max_fragment_length` in each call to `next_tls_record/3`? @IngelaAndin

Reply:
It seems this is used to propagate the value of the negotiated max_fragment_length tls_record:validate_tls_record_length/7 without adding an extra argument to all relevant functions. I think it makes the code confusing!